nuleaf-source/controllers/posts.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * Posts Controller
+ * 
+ * This controller handles requests for creating, updating, deleting, and searching 
+ * posts. All responses are in JSON format.
+ */
+
+/* jslint node: true */
+'use strict';
+
+var PostsDAO = require('..database/posts');
+
+
+/**
+ * Search for posts in the database. A successful response will always
+ * be an array even if no steminars were found matching the filters.
+ *
+ * Parameters list:
+ *     title           : Filter for posts with this title.
+ *     content         : Filter for posts with this content.
+ *     author          : Filter for posts with this author.
+ *     sort            : Stores a number that determines if the results are shown in de/ascending order.
+ *     sortBy          : Stores the attribute above that the results are sorted by.
+ *     skip            : Return a certain number of results after a certain number of documents.
+ *     limit           : Used to specify the maximum number of results to be returned.
+ *     created_before  : Filter for posts after this date.
+ *     created_after   : Filter for posts before this date.
+ *     published_before: Filter for posts after this date.
+ *     published_after : Filter for posts before this date.
+ *     modified_before : Filter for posts after this date.
+ *     modified_after  : Filter for posts before this date.
+ */
+exports.search = function(req, res) {
+  PostsDao.find({
+    title         : req.query.title,
+    content       : req.query.content,
+    author        : req.query.author,
+    sort          : req.query.sort,
+    sortBy        : req.query.sortBy,
+    skip          : req.query.skip,
```

Comment:
You don't have to do this in this PR, but the case when skip and limit is empty should be handled. In such cases, a default should be given. A [short circuit evaluation](http://stackoverflow.com/questions/2100758/javascript-or-variable-assignment-explanation) is probably best for this.